crates/red_knot_python_semantic/src/semantic_index/definition.rs

```diff
@@ -224,12 +226,17 @@ impl DefinitionNodeRef<'_> {
             DefinitionNodeRef::NamedExpression(named) => {
                 DefinitionKind::NamedExpression(AstNodeRef::new(parsed, named))
             }
-            DefinitionNodeRef::Assignment(AssignmentDefinitionNodeRef { assignment, target }) => {
-                DefinitionKind::Assignment(AssignmentDefinitionKind {
-                    assignment: AstNodeRef::new(parsed.clone(), assignment),
-                    target: AstNodeRef::new(parsed, target),
-                })
-            }
+            DefinitionNodeRef::Assignment(AssignmentDefinitionNodeRef {
+                value,
+                target,
+                variable,
+                kind,
+            }) => DefinitionKind::Assignment(AssignmentDefinitionKind {
+                value: AstNodeRef::new(parsed.clone(), value),
+                target: AstNodeRef::new(parsed.clone(), target),
+                variable: AstNodeRef::new(parsed, variable),
```

Comment:
Creating `AstNodeRef`'s isn't expensive but it also isn't free because it requires bumping an `Arc` (and decrementing it later). 

It might be worth to keep storing the assignment only with an index for the right target. I'm not sure what `variable` is for.

Reply:
> It might be worth to keep storing the assignment only with an index for the right target. I'm not sure what `variable` is for.

(lol, after writing a couple of sentences I realize which target you're referring to.)

Ah, yes, I think that can be done.

The `variable` is used to find the symbol the definition belongs to in case the LHS is a sequence. It is also used to add the "symbol -> type" mapping in `self.types.expressions` hashmap.

Reply:
Using indices is quite tricky because of arbitrary nesting of unpacking; this isn't like imports where it's a flat list of items. It's possible but requires keeping a counter across the entire recursive traversal; I think the current code is a lot clearer.

I was unclear about `variable` at first also; I think it is needed for the unique per-Definition NodeKey. Perhaps `name` would be clearer than `variable` -- I think it is always a `Name` node.

Reply:
> Using indices is quite tricky because of arbitrary nesting of unpacking; this isn't like imports where it's a flat list of items. It's possible but requires keeping a counter across the entire recursive traversal; I think the current code is a lot clearer.

I _think_ Micha is talking about the `targets` list in a multi-assignment statement like `x = y = 1` in which case we could just store the single `AstNodeRef<ast::StmtAssign>` and a `usize` index into the `targets` list. @MichaReiser can confirm though

> I was unclear about `variable` at first also; I think it is needed for the unique per-Definition NodeKey. Perhaps `name` would be clearer than `variable` -- I think it is always a `Name` node.

Yes, the `name` field already existed by the name of `target`, I mainly renamed it to variable to avoid any clash with the `targets` field in the node itself. I think `name` is more clear.